src/core.jl

```diff
@@ -0,0 +1,170 @@
+"""
+Type representing any object drawable on image
+"""
+abstract Drawable
+
+"""
+    p = Point(x,y)
+    p = Point(c)
+
+A `Drawable` point on the image
+"""
+immutable Point <: Drawable
+    x::Int
+    y::Int
+end
+
+abstract Line <: Drawable
+abstract Circle <: Drawable
+
+
+"""
+    line = LineTwoPoints(p1, p2)
+
+A `Drawable` infinite length line passing through the two points
+`p1` and `p2`.
+"""
+immutable LineTwoPoints <: Line
+    p1::Point
+    p2::Point
+end
+
+"""
+    line = LineNormal(ρ, θ)
+
+A `Drawable` infinte length line having perpendicular length `ρ` from
+origin and angle `θ` between the perpendicular and x-axis
+"""
+immutable LineNormal <: Line
+    ρ::Real
+    θ::Real
```

Comment:
Well, the thing you are testing (i.e. the constructor) isn't really the bottleneck I was thinking off. I am more worried about the algorithm that use those variables in a loop multiple times. I'll point one such position out to you with a separate comment.